### ISSUE TITLE
cursor.get_cursor should return self

### ISSUE THREAD

**Issue by chadwhitacre** [chadwhitacre/postgres.py#40]
That will make it easier to interchange db and cursor objects. Cropped up at https://github.com/gratipay/gratipay.com/issues/2729#issuecomment-54536544. Similar to #39.

**Comment by Changaco** [comment]
I don't think this is a good idea, because it makes it easy to introduce subtle issues. Requiring explicit changes to support both DB and cursor objects in a function seems less likely to result in bugs.

A method that explicitly supports DB and cursor objects can look like this:

```python
def foo(self, cursor=None):
    (self.db or cursor).run("...")
```

or that:

```python
def foo(self, cursor=None):
    with self.db.get_cursor(cursor=cursor) as c:
        ...
```

This second pattern currently isn't supported by `postgres.py`, it's [implemented in Liberapay](https://github.com/liberapay/liberapay.com/blob/444/liberapay/models/__init__.py#L123-L128) by subclassing `Postgres`.
